### PR TITLE
Add /proc/ish

### PR DESCRIPTION
**Pull request by saagarjha**
Create /proc/ish and populate /proc/ish/version to start with.

### CODE REVIEW

File: fs/proc/ish.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef FS_PROC_ISH_H
+#define FS_PROC_ISH_H
+
+int proc_ish_show_version(struct proc_entry *UNUSED(entry), struct proc_data *buf);
+
+#define PROCFS_ISH_ENTRIES \
```
Comment (tbodt) [inline fs/proc/ish.h:6]: Why a macro? Should just be in ish.c

File: fs/proc/ish.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    {"version", .show = proc_ish_show_version},
+
+#define PROCFS_ISH_ENTRY_COUNT (sizeof((struct proc_dir_entry []) { PROCFS_ISH_ENTRIES }) / sizeof(struct proc_dir_entry))
```
Comment (tbodt) [inline fs/proc/ish.h:9]: This sucks

File: fs/proc/ish.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+struct proc_dir_entry proc_ish_entries[PROCFS_ISH_ENTRY_COUNT];
+
+#define PROCFS_ISH_ENTRY {"ish", S_IFDIR, .children = proc_ish_entries, .children_sizeof = sizeof(proc_ish_entries)}
```
Comment (tbodt) [inline fs/proc/ish.h:13]: Same here, though I can somewhat more understand why this macro

**Comment by saagarjha** [comment]
Yeah, all of these suck but they're necessary to make the `proc_ish_entries` be defined with the correct size. I think this is the best option among several poor ones.

**Pull request merged**